src/fault_injector/fault_types/train_prio_fault.py

```diff
@@ -39,6 +40,12 @@ def resolve_fault(self, tick: int):
         :param tick: the simulation tick in which resolve_fault was called
         :type tick: Integer
         """
-        self.train.train_type.priority = self.old_prio
-        self.interlocking.insert_train_priority_changed(self.train)
-        self.logger.resolve_train_prio_fault(tick, self.configuration.id)
+        if self.train is None:
+            raise ValueError("Fault not injected")
+
+        if self.train is self.get_train(
```

Comment:
If the train left the simulation an error would be raised here, as ```get_train``` is called. Maybe you need a new method here, that returns if a train is still in the simulation or not.